Content.Server/Body/Systems/InternalsSystem.cs

```diff
@@ -38,6 +40,38 @@ public override void Initialize()
         SubscribeLocalEvent<InternalsComponent, ComponentShutdown>(OnInternalsShutdown);
         SubscribeLocalEvent<InternalsComponent, GetVerbsEvent<InteractionVerb>>(OnGetInteractionVerbs);
         SubscribeLocalEvent<InternalsComponent, InternalsDoAfterEvent>(OnDoAfter);
+        SubscribeLocalEvent<BreathToolComponent, ConnectedBreathToolEvent>(OnBreathMaskConnected);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var query = EntityQueryEnumerator<InternalsDelayedActivationComponent>();
+        while (query.MoveNext(out var uid, out var delayComp))
+        {
+            if (_timing.CurTime < delayComp.Time)
+                return;
+
+            if (TryComp<InternalsComponent>(uid, out var comp))
+                ToggleInternals(delayComp.Entity , delayComp.Entity , false, comp);
+
+            RemCompDeferred<InternalsDelayedActivationComponent>(uid);
+        }
+    }
+
+    private void OnBreathMaskConnected(EntityUid uid, BreathToolComponent comp, ConnectedBreathToolEvent args)
+    {
+        if (!comp.AutomaticActivation)
+            return;
+
+        // delaying until next tick, because at spawn we can't be sure the gas tank will spawn before the mask
+        EnsureComp<InternalsDelayedActivationComponent>(args.User, out var delayComp);
+        delayComp.Time= _timing.CurTime + TimeSpan.FromTicks(1);
+        delayComp.Entity = args.User;
+
+        // this is currently only used for roundstart activation, so we turn the feature off afterwards
+        comp.AutomaticActivation = false;
     }
```

Comment:
Isn't this going to be really janky in gameplay? It's also not predicted.

Reply:
I'm not sure how/which part of this is janky gameplay-wise? Can you please be more specific

Reply:
Well the one part I see as janky, currently, is that there is a doafter for the internal turning on. But I didn't want to duplicate parts of the toggle code to be able to do it without the doafter. I'm thinking I'll put an "Instant" bool there that allows calling the same classes and just skips over the doafter

Reply:
The doafter for toggling internals was removed on master so that's no longer an issue for auto-activation either

Reply:
It should also be predicted now